spock-core/src/main/java/org/spockframework/runtime/Condition.java

```diff
@@ -115,8 +119,23 @@ private void createRendering() {
     rendering = builder.toString();
   }
 
-  private String flatten(String text) {
-    return pattern.matcher(text).replaceAll(" ");
+  private static CharSequence stripAndFlatten(String text) {
+    Matcher m = pattern.matcher(text);
+    StringBuffer sb = new StringBuffer();
```

Comment:
The problem is that ```java.util.regex.Matcher``` uses ```StringBuffer``` in ```appendReplacement``` and ```appendTail``` methods. I think we could rework methods so that it doesn't use these handy methods on ```Matcher``` but don't know how exactly for now. I will need some time to think about it if we don't want to use ```StringBuffer``` here